src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -208,19 +199,30 @@ private TrieNode searchNode(byte[] bytes, int offset, int length) {
     }
 
     @CheckForNull
-    private TrieNode searchForJsonPathKeyNode(byte[] bytes, Iterator<? extends JsonPathSegmentReference> jsonPath) {
+    private TrieNode searchForJsonPathKeyNode(byte[] bytes, Iterator<? extends JsonPathNode> jsonPath) {
         TrieNode node = root;
         node = node.children['$' + BYTE_OFFSET];
         if (node == null) {
             return null;
         }
+        if (node.endOfWord) {
```

Comment:
I think this can be moved inside while tk be the very first statement, this way you won't need to additionally check in wildcard branch

Reply:
Hm, or maybe it's actually wrong? If you would have different configuration for `$` and `$.a`, then we should've used latter and do not short circuit

Reply:
I banned this type of overlapping keys `$` and `$.a`, so there shouldn't be an issue. But there could be cases like `$.ab` and `$.a`, in this case checking for `endOfWord` in the beginning of the loop would be wrong.